gateway/modules/eventing/eventing.go

```diff
@@ -118,6 +125,48 @@ func (m *Module) SetConfig(project string, eventing *config.Eventing) error {
 	if m.config.InternalRules == nil {
 		m.config.InternalRules = map[string]config.EventingRule{}
 	}
+	// err := m.CloseConfig(project, eventing)
+	// if err != nil {
+	// 	fmt.Printf("setconfig\n")
+	// }
+	return nil
+}
 
+// CloseConfig closes the module config
+func (m *Module) CloseConfig(project string, eventing *config.Eventing) error {
+	m.eventChanMap = sync.Map{}
+	m.processingEvents = sync.Map{}
+	//erase map
+	m.eventChanMap.Range(func(key interface{}, value interface{}) bool {
+		m.eventChanMap.Delete(key)
+		return true
+	})
+	//erase map
+	m.processingEvents.Range(func(key interface{}, value interface{}) bool {
+		m.processingEvents.Delete(key)
+		return true
+	})
+
+	for k := range m.schemas {
+		delete(m.schemas, k)
+	}
+	for k := range eventing.Rules {
+		delete(eventing.Rules, k)
+	}
+	for k := range eventing.InternalRules {
+		delete(eventing.InternalRules, k)
+	}
+	for k := range eventing.SecurityRules {
+		delete(eventing.SecurityRules, k)
+	}
+	for k := range eventing.Schemas {
+		delete(eventing.Schemas, k)
+	}
+	//m.stopChan = make(chan struct{})
+	fmt.Println("main: telling goroutines to stop")
+	close(m.stopChan)
```

Comment:
i believe this will not work, you are directly closing the channel. 
as your go routine has a default case :
reciving on a closed channel will never occur

figure out some other way use map or array